### PR TITLE
10336-android-validate-event-name

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
https://swrvedev.jira.com/browse/SWRVE-10336

@Sergio-Mira @shanetmoore please

### CODE REVIEW

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 1]
```diff
@@ -1372,7 +1372,9 @@ public void sessionEnd() {
     @Override
     public void event(String name) {
         try {
-            _event(name);
+            if(isValidEventName(name)) {
+                _event(name);
+            }
```
Comment (shanetmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Can we add a logger statement here to notify the user that they're event couldn't be sent due to it starting with Swrve?

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 2]
```diff
@@ -1381,12 +1383,29 @@ public void event(String name) {
...
             SwrveLogger.e(LOG_TAG, "Exception thrown in Swrve SDK", e);
         }
     }
 
+    private boolean isValidEventName(String name) {
+        List<String> restrictedNamesStartWith = new ArrayList<String>() {{
+            add("Swrve.");
+            add("swrve.");
+        }};
+
```
Comment (shanetmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Not a requirement but this is a handy one I use for list initialization: ` List<String> restrictedNamesStartWith = Arrays.asList("Swrve.", "swrve.");`

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java (hunk 1 above)
Comment (shanetmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Oh nvm I see the logger statement in the isValidEvent method

Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: :+1:

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java (hunk 2 above)
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Yup, knew that. Thought I'd go for this instead. :+1:

**Pull request merged**